crates/graphql/src/schema.rs

```diff
@@ -88,17 +121,74 @@ impl<'a> From<models::Storefront<'a>> for Storefront {
     }
 }
 
-#[derive(GraphQLInputObject)]
-#[graphql(description = "Buy a NFT")]
-struct BuyNft {
-    transaction: String,
-}
-
 pub struct QueryRoot {
     db: Pool,
 }
 
-#[juniper::graphql_object]
+pub struct NftDetailBatcher {
+    db_connection: PooledConnection,
+}
+
+#[async_trait]
+impl BatchFn<String, Option<NftDetail>> for NftDetailBatcher {
+    async fn load(&mut self, keys: &[String]) -> HashMap<String, Option<NftDetail>> {
+        let conn = &self.db_connection;
+        let mut hash_map = HashMap::new();
+
+        for key in keys {
+            hash_map.insert(key.clone(), None);
+        }
```

Comment:
While building this out for John Le there was 1 NFT that didn't have an associated NFT so the load call would panic. This is in t default every key to None and then override. Now that I review thinking the best is to capture the panic in the load and return none there. Going to look into that any start on capturing the error welcome.